api/filters/replacement/replacement.go

```diff
@@ -93,7 +98,24 @@ func getRefinedValue(options *types.FieldOptions, rn *yaml.RNode) (*yaml.RNode,
 	if rn.YNode().Kind != yaml.ScalarNode {
 		return nil, fmt.Errorf("delimiter option can only be used with scalar nodes")
 	}
-	value := strings.Split(yaml.GetValue(rn), options.Delimiter)
+	value := []string{}
+	if options.EndDelimiter == "" {
+		value = strings.Split(yaml.GetValue(rn), options.Delimiter)
+	} else {
+		mapper := func(s string) string {
+			s = strings.ReplaceAll(s, options.Delimiter, "")
+			s = strings.ReplaceAll(s, options.EndDelimiter, "")
+			return s
+		}
+		if options.Delimiter == "" {
+			return nil, fmt.Errorf("delimiter needs to be set if enddelimiter is set")
+		}
+		re := regexp.MustCompile(regexp.QuoteMeta(options.Delimiter) + `(.*?)` + regexp.QuoteMeta(options.EndDelimiter))
```

Comment:
This seems to be a bit fragile:
- We assume that delimiter and end delimiter are in pairs, but this may not be the case always. 
- Also what happens when delimiters are not in expected order, or there are nested delimiters. 

If not for nested delimiters, for the other cases, can we do any kind of pre-processing?

Reply:
@varshaprasad96 thank you for the feedback. 

- The option is that `Delimiter` comes alone, in this case it stays like it is today, the `EndDelimiter` should not be used without a `Delimiter` if so we return an error and if they are used in combination, the value in between them is loaded. If I'm not mistaken all cases should be handled.
- The delimiters should always be in expected order because their key (delimiter or enddelimiter) allow us to define the order.
- To make it less fragile we could do a preprocessing, if the `EndDelimiter` is set and the `Delimiter` is empty they swap values. Do you have another suggestion for preprocessing?

Reply:
@varshaprasad96 I will look into this this week and give you a feedback. If I see a possibility to ensure the order and the amount of delimiters I would implement it so you can review it.

Reply:
Hi @varshaprasad96 I've looked into it and wanted to ask if you could provide a nested example for me. As I have seen the delimiters are used in `FieldOptions` https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/replacement.go#L67 which contains of primitive datatypes variables, therefore I don't see how multiple delimiters or nesting could accure. The replacement consists of one `SourceSelector` and multiple `TargetSelector` everyone of these types contain exactly one `FieldOptions` type. 

If I am missing something just tell me and I will look into it.